apps/gnome-mobile/src/app/(app)/(tabs)/index.tsx

```diff
@@ -117,49 +195,53 @@ const MapScreen = () => {
         minZoomLevel={18}
         maxZoomLevel={20}
         ref={ref}
+        onRegionChangeComplete={(region) => {
+          if (
+            region.latitude !== userLocation.latitude ||
+            region.longitude !== userLocation.longitude
+          ) {
+            ref.current?.animateToRegion(
+              {
+                latitude: userLocation.latitude,
+                longitude: userLocation.longitude,
+                latitudeDelta: defaultRegion.latitudeDelta,
+                longitudeDelta: defaultRegion.longitudeDelta,
+              },
+              100,
+            );
+          }
+        }}
       >
-        <Marker
-          coordinate={{
-            latitude: 51.09701966184086,
-            longitude: 17.035843526079727,
-          }}
-          title="Krasnal Podróżnik"
-          description="Krasnal wysiadający z autobusu"
-        >
-          <Image
-            source={require("@/assets/images/krasnal.png")}
-            style={{ width: 30, height: 30 }}
-          />
-        </Marker>
-        <Marker
-          coordinate={{
-            latitude: 51.10894028789954,
-            longitude: 17.03288804137201,
-          }}
-          title="Krasnale Syzyfki"
-          description="Ciągle pchają ten nieszczęsny kamień"
-        >
-          <Image
-            source={require("@/assets/images/krasnal.png")}
-            style={{ width: 30, height: 30 }}
-          />
-        </Marker>
-        <Marker
-          coordinate={{
-            latitude: 51.10947379220885,
-            longitude: 17.057842134960516,
-          }}
-          title="Krasnal Mędruś"
-          description="Ponoć studenci przychodzą do niego po porady"
-        >
-          <Image
-            source={require("@/assets/images/krasnal.png")}
-            style={{ width: 30, height: 30 }}
-          />
-        </Marker>
+        {markers.map((marker, index) => (
+          <Marker
+            key={index}
+            coordinate={{
+              latitude: marker.latitude,
+              longitude: marker.longitude,
+            }}
+            title={marker.title}
+            description={marker.description}
+          >
+            <Image
+              source={require("@/assets/images/krasnal.png")}
+              style={{ width: 30, height: 30 }}
+            />
+          </Marker>
+        ))}
       </MapView>
 
       <HeaderControls user={user} replace={replace} />
+
+      <View className="absolute bottom-0 left-0 right-0 p-4 bg-transparent">
+        {reachedMarker ? (
+          <DraggableGnome onUnlock={() => Alert.alert("Gnome Unlocked!")} />
+        ) : (
+          <DistanceTracker
+            distance={distance}
+            showDistanceTracker={showDistanceTracker}
+          />
+        )}
```

Comment:
spoko opcja ale mam tu propozycję żeby było to jeszcze czytelniejsze

ogólnie to zamiast dawać zmienną showDistanceTracker która w środku DistanceTrackera powoduje że nic nie jest wyświetlone to lepiej tej zmiennej użyć tutaj

```suggestion
        {reachedMarker 
          ? <DraggableGnome onUnlock={() => Alert.alert("Gnome Unlocked!")} />
          : (showDistanceTracker 
            ? <DistanceTracker distance={distance} /> 
            : null
        ))}
```